client/src/components/profile/components/portfolio.tsx

```diff
@@ -55,95 +49,89 @@ function createFindById(id: string) {
   return (p: PortfolioProjectData) => p.id === id;
 }
 
-class PortfolioSettings extends Component<PortfolioProps, PortfolioState> {
-  validationImage: HTMLImageElement;
-  static displayName: string;
-  constructor(props: PortfolioProps) {
-    super(props);
-    this.validationImage = new Image();
-    const { portfolio = [] } = props;
+const PortfolioSettings = (props: PortfolioProps) => {
+  const {
+    t,
+    portfolio: initialPortfolio = [],
+    setIsEditing,
+    updatePortfolio
+  } = props;
+  const [portfolio, setPortfolio] = useState(initialPortfolio);
+  const [unsavedItemId, setUnsavedItemId] = useState<string | null>(null);
+  const [imageValidation, setImageValid] = useState<ProfileValidation>({
+    state: 'success',
+    message: ''
+  });
+
+  const checkIfValidImage = (url: string): Promise<ProfileValidation> => {
+    const img = new Image();
 
-    this.state = {
-      portfolio: [...portfolio],
-      unsavedItemId: null,
-      isImageValid: {
-        state: 'success',
-        message: ''
-      }
-    };
-  }
-
-  toggleEditing = () => {
-    this.props.setIsEditing(false);
+    return new Promise(resolve => {
+      img.onerror = () =>
+        resolve({ state: 'error', message: t('validation.url-not-image') });
+      img.onload = () => resolve({ state: 'success', message: '' });
+      img.src = url;
+    });
```

Comment:
I recommend sharing the original `checkIfValidImage` function that's currently in `about.tsx`. The logic is common to the two approaches, so we might as well share the code. We already have client/src/components/profile/components/utils/utils.ts which seems like the obvious home for `checkIfValidImage`